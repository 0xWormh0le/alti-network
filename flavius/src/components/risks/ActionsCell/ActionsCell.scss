.ActionsCell {
  min-width: 85px;
  max-width: 300px;
  position: relative;
}
