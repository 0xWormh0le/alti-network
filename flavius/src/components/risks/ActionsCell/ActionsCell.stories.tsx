import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import ActionsCell from './ActionsCell'
import { BrowserRouter } from 'react-router-dom'

export default {
  title: 'Risks/ActionsCell',
  component: ActionsCell
} as Meta

const Template: Story<any> = (args) => (
  <BrowserRouter>
    <ActionsCell {...args} />
  </BrowserRouter>
)

const fileData = new Map()
fileData.set('fileCount', 10)

export const Default = Template.bind({})
Default.args = {}

export const WithData = Template.bind({})
WithData.args = {
  email: 'bobbie@thoughtlabs.io',
  file: fileData,
  appName: 'foobar',
  appId: 'baz',
  riskId: '0',
  managePermissionsActionEnabled: true,
  riskTypeId: 10,
  emailActionEnabled: true,
  owner: {
    get: (x: any) => x
  }
}
