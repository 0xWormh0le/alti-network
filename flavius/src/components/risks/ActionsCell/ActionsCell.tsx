import React from 'react'
import cx from 'classnames'
import { modalBasePath, searchWithoutModalParams } from 'util/helpers'
import DropdownButton, { ActionType } from 'components/elements/DropdownButton/DropdownButton'
import UI_STRINGS from 'util/ui-strings'
import './ActionsCell.scss'

const NOT_ENABLED_MESSAGE = 'not enabled in this version'
const NOT_ENABLED_FOR_THIS_RISK_MESSAGE = 'not enabled for this type of Risk'

export type ActionCallback = (...args: any[]) => void

interface ActionsCellProps {
  onEmailAction: ActionCallback
  onManagePermissionsAction: ActionCallback
  onLockoutAction: ActionCallback
  onIgnoreRiskAction: ActionCallback
  onPutBackToActiveAction: ActionCallback
  state: UserVisibilityState
  emailActionEnabled: boolean
  managePermissionsActionEnabled: boolean
  file?: any
  riskId?: string
  email?: string
  riskTypeId?: number
  appId?: string
  appName?: string
  owner?: IPerson
  platformId?: any
}

const ActionsCell: React.FC<ActionsCellProps> = ({
  onEmailAction,
  emailActionEnabled,
  onManagePermissionsAction,
  managePermissionsActionEnabled,
  onLockoutAction,
  onIgnoreRiskAction,
  onPutBackToActiveAction,
  state,
  file,
  riskId,
  email,
  riskTypeId,
  appId,
  appName,
  platformId,
  owner
}) => {
  const ownerEmail: string = owner?.primaryEmail?.address || ''
  const fileCount: number = file.fileCount
  const singleFileRisk: boolean = fileCount === 1

  const resolveRiskLink = {
    pathname: `${modalBasePath()}/resolve/${riskId}`,
    search: searchWithoutModalParams({
      riskId,
      riskTypeId,
      email: email ? email : ownerEmail,
      appName,
      appId,
      fileCount,
      platformId
    })
  }

  const actions: ActionType[] = [
    {
      onClick: onEmailAction,
      title: UI_STRINGS.RISKS.ALERT_FILE_OWNER,
      actionEnabled: emailActionEnabled,
      link: ''
    },
    {
      onClick: () => {
        onManagePermissionsAction()
      },
      title: UI_STRINGS.RISKS.RESOLVE_RISK,
      actionSecondaryText: managePermissionsActionEnabled ? '' : NOT_ENABLED_FOR_THIS_RISK_MESSAGE,
      actionEnabled: managePermissionsActionEnabled || singleFileRisk,
      link: resolveRiskLink
    },
    {
      onClick: onLockoutAction,
      title: UI_STRINGS.RISKS.LOCK_DOCUMENT,
      actionSecondaryText: NOT_ENABLED_MESSAGE,
      actionEnabled: false,
      link: ''
    },
    {
      onClick: onIgnoreRiskAction,
      title: UI_STRINGS.RISKS.IGNORE_RISK,
      actionEnabled: state === 'active',
      link: ''
    },
    {
      onClick: onPutBackToActiveAction,
      title: UI_STRINGS.RISKS.PUT_BACK_AS_ACTIVE,
      actionEnabled: state === 'ignored',
      link: ''
    }
  ]

  return (
    <div className={cx('ActionsCell')}>
      <DropdownButton text={UI_STRINGS.DROPDOWN.ACTIONS} actions={actions} enabled={true} />
    </div>
  )
}

export default ActionsCell
