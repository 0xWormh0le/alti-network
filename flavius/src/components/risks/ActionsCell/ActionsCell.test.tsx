import React from 'react'
import { render } from '@testing-library/react'
import ActionsCell, { ActionCallback } from './ActionsCell'

let onEmailAction: ActionCallback
let onManagePermissionsAction: ActionCallback
let onLockoutAction: ActionCallback
let onIgnoreRiskAction: ActionCallback
let onPutBackToActiveAction: ActionCallback

beforeEach(() => {
  onEmailAction = jest.fn()
  onManagePermissionsAction = jest.fn()
  onLockoutAction = jest.fn()
  onIgnoreRiskAction = jest.fn()
  onPutBackToActiveAction = jest.fn()
})

it('renders correctly', () => {
  const fileData = new Map()
  fileData.set('fileCount', 10)
  const { container } = render(
    <ActionsCell
      onPutBackToActiveAction={onPutBackToActiveAction}
      onIgnoreRiskAction={onIgnoreRiskAction}
      onEmailAction={onEmailAction}
      state='active'
      emailActionEnabled={false}
      onManagePermissionsAction={onManagePermissionsAction}
      managePermissionsActionEnabled={false}
      onLockoutAction={onLockoutAction}
      file={fileData}
    />
  )
  expect(container).toMatchSnapshot()
})

it('should render 0 actions', () => {
  const fileData = new Map()
  fileData.set('fileCount', 10)
  const { container } = render(
    <ActionsCell
      onPutBackToActiveAction={onPutBackToActiveAction}
      onIgnoreRiskAction={onIgnoreRiskAction}
      onEmailAction={onEmailAction}
      state='active'
      emailActionEnabled={false}
      onManagePermissionsAction={onManagePermissionsAction}
      managePermissionsActionEnabled={false}
      onLockoutAction={onLockoutAction}
      file={fileData}
    />
  )
  expect(container.querySelectorAll('.ActionCell__popup--item').length).toBe(0)
})
