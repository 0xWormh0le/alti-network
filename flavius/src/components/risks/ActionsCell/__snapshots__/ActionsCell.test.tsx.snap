// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="ActionsCell"
  >
    <div
      class="DropdownWrapper"
      tabindex="1"
    >
      <button
        class="Button Button--dropdown Button--size-medium Button--secondary"
        type="button"
      >
        Actions
      </button>
      <svg
        class="DropdownPopup--arrow-off"
        height="8px"
        version="1.1"
        viewBox="0 0 12 8"
        width="12px"
      >
        <polygon
          id="path-1"
          points="6 8 12 0 0 0"
        />
      </svg>
      <ul
        class="DropdownPopup"
      >
        <span
          class="DropdownPopup--arrow"
        />
        
        
        
        <li
          class="DropdownPopup--item"
        >
          <span>
            Ignore Risk
          </span>
        </li>
        
      </ul>
    </div>
  </div>
</div>
`;
